src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFile.cs

```diff
@@ -128,7 +128,44 @@ public bool Equals(LockFile other)
                 && Libraries.OrderedEquals(other.Libraries, library => library.Name, StringComparer.OrdinalIgnoreCase)
                 && Targets.OrderedEquals(other.Targets, target => target.Name, StringComparer.Ordinal)
                 && PackageFolders.SequenceEqual(other.PackageFolders)
-                && EqualityUtility.EqualsWithNullCheck(PackageSpec, other.PackageSpec);
+                && EqualityUtility.EqualsWithNullCheck(PackageSpec, other.PackageSpec)
+                && LogsEqual(other.LogMessages);
+        }
+
+        private bool LogsEqual(IList<IAssetsLogMessage> otherLogMessages)
+        {
+            if(LogMessages == otherLogMessages)
+            {
+                return true;
+            }
+            if(LogMessages.Count != otherLogMessages.Count)
+            {
+                return false;
+            }
+
+
+            var equals = true;
+
+            var orderedLogMessages = LogMessages.OrderBy(m => m.Message, StringComparer.Ordinal)
+                .ToArray();
+
+            var orderedOtherLogMessages = otherLogMessages.OrderBy(m => m.Message, StringComparer.Ordinal)
+                .ToArray();
+
+            var length = orderedLogMessages.Length;
+
+            for(var i=0; i<length; i++)
+            {
+                equals &= orderedLogMessages[i].LogEquals(orderedOtherLogMessages[i]);
```

Comment:
Consider using .Zip from linq to combine two lists and return a bool for each

https://msdn.microsoft.com/en-us/library/dd267698(v=vs.110).aspx